shared/components/DemoApp/Error404/index.js

```diff
@@ -15,10 +15,9 @@ class Error404 extends Component {
   }
 }
 
-Error404.contextTypes = {
-  history: PropTypes.shape({
-    staticContext: PropTypes.object,
-  }).isRequired,
+Error404.propTypes = {
+  // eslint-disable-next-line react/forbid-prop-types
+  staticContext: PropTypes.object.isRequired,
```

Comment:
No, It should not. I prefer not to provide default value for `staticContext` prop.